### ISSUE TITLE
1.08: Tests fail

### ISSUE THREAD

**Issue by eserte** [jonasbn/perl-date-holidays#21]
# Issue Template

### Description

Fail log is very long. It starts with

```
Unable to load module: Date::Holidays::Adapter::Supered - Unable to load: Date::Holidays::Adapter::Supered - Can't locate Date/Holidays/Adapter/Supered.pm in @INC (@INC contains: /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/t/../lib /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/t/../t /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/t/t /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/t /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/blib/lib /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/blib/arch /usr/perl5.16.3t/lib/site_perl/5.16.3/amd64-freebsd-thread-multi /usr/perl5.16.3t/lib/site_perl/5.16.3 /usr/perl5.16.3t/lib/5.16.3/amd64-freebsd-thread-multi /usr/perl5.16.3t/lib/5.16.3 .) at /usr/perl5.16.3t/lib/site_perl/5.16.3/Module/Load.pm line 77.
Can't locate Date/Holidays/Adapter/Supered in @INC (@INC contains: /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/t/../lib /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/t/../t /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/t/t /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/t /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/blib/lib /usr/home/eserte/.cpan/build/2018052821/Date-Holidays-1.08-2/blib/arch /usr/perl5.16.3t/lib/site_perl/5.16.3/amd64-freebsd-thread-multi /usr/perl5.16.3t/lib/site_perl/5.16.3 /usr/perl5.16.3t/lib/5.16.3/amd64-freebsd-thread-multi /usr/perl5.16.3t/lib/5.16.3 .) at /usr/perl5.16.3t/lib/site_perl/5.16.3/Module/Load.pm line 77.
```

# Bugs and Issues

### Specify what version was used

1.08

### Supply some information on the environment

Various Perls: 5.8.9 .. 5.27.11
Various OS: FreeBSD (9, 10, 11, 12), Debian Linux (jessie - stretch), CentOS 6

**Comment by jonasbn** [comment]
Hi @eserte 
Is this from CPAN-testers? or could you provide some information on how you execute? It looks as if the `t/` directory is included in `@INC` for the test

**Comment by jonasbn** [comment]
CPAN-testers reports: http://www.cpantesters.org/distro/D/Date-Holidays.html?grade=3&perlmat=2&patches=2&oncpan=2&distmat=2&perlver=ALL&osname=ALL&version=1.08

**Comment by eserte** [comment]
It does not matter if I run it with cpan (i.e. `cpan -t Date::Holidays`) or run `make test` in the extracted tarball --- I always get the fails in t/runtests.t.

**Comment by jonasbn** [comment]
This [particular test report](http://www.cpantesters.org/cpan/report/a98852e0-62f8-11e8-9c4b-c48231c3c7d0) indicating failure also describes the issue (broken up into separate lines for clarity).

```
Unable to load module: Date::Holidays::Adapter::Supered - Unable to load: Date::Holidays::Adapter::Supered - Can't locate Date/Holidays/Adapter/Supered.pm in @INC (@INC contains: 
/tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/t/../lib 
/tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/t/../t 
/tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/t/t 
/tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/t 
/tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/blib/lib 
/tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/blib/arch 
/opt/perl-5.8.9/lib/5.8.9/x86_64-linux 
/opt/perl-5.8.9/lib/5.8.9 
/opt/perl-5.8.9/lib/site_perl/5.8.9/x86_64-linux /opt/perl-5.8.9/lib/site_perl/5.8.9 .) at /opt/perl-5.8.9/lib/site_perl/5.8.9/Module/Load.pm line 77.
Can't locate Date/Holidays/Adapter/Supered in @INC (@INC contains: /tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/t/../lib /tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/t/../t /tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/t/t /tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/t /tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/blib/lib /tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/blib/arch /opt/perl-5.8.9/lib/5.8.9/x86_64-linux /opt/perl-5.8.9/lib/5.8.9 /opt/perl-5.8.9/lib/site_perl/5.8.9/x86_64-linux /opt/perl-5.8.9/lib/site_perl/5.8.9 .) at /opt/perl-5.8.9/lib/site_perl/5.8.9/Module/Load.pm line 77.
```

In `t/runtests.t` the following lines are defining the additional include path (`@INC`) for use by the test suite:

```perl
use FindBin qw($Bin);
use lib "$Bin/../lib", "$Bin/../t", "$Bin/t", "$Bin";
```

And since:

```
/tmpfs/.cpan-build-cpansand/2018052821/Date-Holidays-1.08-0/t
```

 is listed in the above report and hence it should be included - which is makes the fail a bit confusing

**Comment by jonasbn** [comment]
Ah, I think I have found the reason, there is no `Date::Holidays::Adapter::Supered` in the distribution and it should not be necessary. But the restructuring of the loading of adapter classes introduced in 1.08 might beg to differ and therefor this demonstrates the bug. 

Not completely satisfied with the fact that I cannot demonstrate this locally :-/

**Comment by jonasbn** [comment]
I have now investigated this and my conclusion is that some of the tests where broken. This rely on actual Date::Holidays::* distributions to be installed, hence you can get very varied results. I have corrected the issues in the test suite and will release a bug fix release. I have created issues for implementation of adapter classes for NZ (#22) and SK (#23). And there seems to be an issue with the integration with [Date::Holidays::ES](https://metacpan.org/pod/Date::Holidays::ES) which require further investigation (#24).

**Comment by jonasbn** [comment]
I have released 1.09, I will observe the incoming test reports for this release, to see if the problem is still present.

**Comment by jonasbn** [comment]
The adjustments/fixes to the test suite have addressed this issue.

Thanks for the bug report.

jonasbn
